### PR TITLE
Changed to different get_oriented_bounding_box method that fits the points better.

### PR DESCRIPTION
**Pull request by lexihaberl**

### CODE REVIEW

File: src/table_plane_extractor_srv.py [hunk 1]
```diff
@@ -5,12 +5,26 @@
...
 
 
+def get_minimum_oriented_bounding_box(o3d_pc):
+    '''
+    Computes the oriented minimum bounding box of a set of points in 3D space.
+    Input: open3d.geometry.PointCloud o3d_pc
+    Output: open3d.geometry.OrientedBoundingBox o3d_bb
+    '''
+    bb_points = np.array(
+        compas_bb.oriented_bounding_box_numpy(np.array(o3d_pc.points)))
```
Comment (jibweb) [inline src/table_plane_extractor_srv.py]: use np.asarray instead of np.array to avoid copying data and reuse the same memory instead

**Comment by jibweb** [comment]
I'm not familiar with COMPAS but we should always think twice before adding dependencies. I would also point out shapely as a library that might be a more fitting dependency in this case?

What is the problem with open3d bounding boxes?

**Comment by lexihaberl** [comment]
The problem with open3d was that their estimation for a minimum volume bounding box was often way off (~ 50% chance), which lead to the table collision object being ~twice as big as the table itself (which meant that the robot didn't find a plan to execute), and also the plane-pop-out objects often got a strange bounding box (but this was less of a problem, as HAF can deal with it). 
Regarding the dependency: i know but the library is not that big (20MBish) so I thought that's still ok.
Regarding shapely: in my quick search I didn't find a method for getting the 3D-BoundingBox.

File: src/table_plane_extractor_srv.py (hunk 1 above)
Comment (lexihaberl) [reply src/table_plane_extractor_srv.py]: Updated

**Comment by jibweb** [comment]
Fair enough! Maybe creating a function similar to [that](https://stackoverflow.com/questions/32892932/create-the-oriented-bounding-box-obb-with-python-and-numpy) in 3D in v4r_util would be worth it? Sorry I am a bit picky, but the purpose of COMPAS (astrophysics something? or am I looking at the wrong one?) makes me think that we will never use it for anything else than that

**Comment by lexihaberl** [comment]
Here is the link to their [website](https://compas.dev/index.html) and [API](https://compas.dev/compas/latest/api/compas.geometry.html)

**Comment by jibweb** [comment]
My bad, I was indeed looking at the wrong library :-D Then yeah, that looks like a valid addition to the dependency list!

**Comment by lexihaberl** [comment]
I pushed the update that moves the function to the v4r-util package and the changes to use np.asarray.

**Comment by jibweb** [comment]
Great! Unless I screwed up with the permission, I think you should also be able to lift the pull request restrictions now :-)

**Review by jibweb — APPROVED** [review]

**Pull request merged**